src/upstream.rs

```diff
@@ -49,7 +49,7 @@ impl Upstream {
         if let Some(extra) = extra {
             let s = s
                 .map(|qs| {
-                    let mut qs = qs.to_owned();
+                    let mut qs = qs.clone();
```

Comment:
@eguzki This change is iffy, I am not 100% sure if is correct.

Reply:
I think it is ok. 

I do not see a good reason to use Cow here, but surely I am missing something.